frontend/app/components/letter-list.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+
+import { Link } from '@remix-run/react';
+
+export interface Letter {
+  id: string;
+  subject: string;
+  dateSent: string;
+  referenceId: string;
+}
+
+export interface LetterListProps {
+  letters: Letter[];
+  i18n: { subject: string; dateSent: string; referenceId: string };
+}
+
+export const LetterList: React.FC<LetterListProps> = ({ letters, i18n }) => {
+  const [isSortedLatest, setIsSortedLatest] = useState<boolean>(true);
+  const [sorted, setSorted] = useState<Letter[]>(letters);
+
+  useEffect(() => {
+    setSorted(sortLetters(letters, isSortedLatest));
+  }, [letters, isSortedLatest]);
+
+  const sortLetters = (letters: Letter[], latest: boolean) => {
+    return [...letters].sort((a, b) => {
+      return latest ? new Date(b.dateSent).getTime() - new Date(a.dateSent).getTime() : new Date(a.dateSent).getTime() - new Date(b.dateSent).getTime();
```

Comment:
Not sure, but I don't think you need to use `getTime()` here.

``` shell
❯ node
Welcome to Node.js v20.10.0.
Type ".help" for more information.
> const a = new Date()
> const b = new Date()
> b - a
4239 # ← 4.2s
```

Reply:
Will get an error if I remove `getTime()`.

`The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.ts(2362)`

However, this doens't break the app.
It's suggesting to use getTime() to converts date object into numeric timestamps, it's more explicit, makes it more accurate.